include/boost/geometry/algorithms/detail/envelope/implementation.hpp

```diff
@@ -43,27 +46,31 @@ namespace detail { namespace envelope
 {
 
 
-struct exterior_ring_expand_policy
-{
-    template <typename Box, typename Geometry>
-    static inline void apply(Box& mbr, Geometry const& geometry)
-    {
-        Box ring_mbr;
-        envelope_range<>::apply(exterior_ring(geometry), ring_mbr);
-        geometry::expand(mbr, ring_mbr);
-    }
-};
-
-
+template <std::size_t Dimension, std::size_t DimensionCount>
```

Comment:
Yes, the example usage can be found in the version for Linear right? I see that this design is generic and I of course like it. I'm just saying that it's slightly confusing because some things have 2 purposes instead of 1. E.g. `disjoint::envelope` and logically "derived" details like `envelope_box` or `envelope_polygon` in your design is used to disptch with both `Dimension` and `CSTag`. Those purposes are not compatible because the 2 first coordinates are handled in the same time in non-cartesian CSes. Therefore IMO `envelope` is not a good place for dispatching with `Dimension`. I'd rather implement all tools working on dimensions ranges deeper and use them in `disjoint::envelope` dispatched only with CS.